File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.java
import java.io.*;
import java.util.*;

public class Main {
        public static void main(String[] args) throws IOException {
            BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
            Set<String> member = new HashSet<>();
            int n = Integer.parseInt(br.readLine());
            int count = 0;
            for (int i = 0; i < n; i++) {
                String temp = br.readLine();
                if (temp.equals("ENTER")) {
                    count += member.size();
                    member.clear();
                } else {
                    member.add(temp);
                }
            }
            count += member.size();
            System.out.print(count);
        }
}

File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.py
def func():
    n = int(input())
    count = 0
    arr = set()
    for i in range(n):
        str = input()
        if (str == "ENTER"):
            arr = set()
            continue
        if (str not in arr):
            arr.add(str)
            count += 1
    return count

if __name__ == "__main__":
    print(f"{func()}")